rust/src/optimize.rs

```diff
@@ -0,0 +1,356 @@
+//! Optimize a Delta Table
+//!
+//! Bin-packing is performed which merges smaller files into a larger file. This
+//! reduces the number of api calls required to read data and allows for the
+//! optimized files to cleaned up by vacuum.
```

Comment:
Two additional things worth mentioning in the docs, IMO:

 1. This increments the table version and removes files, but it does NOT delete old files. To delete old files use vacuum.
 2. It is safe to run concurrently with append-only workloads, but not yet other ones.